bin/scpipe/lib/sc/pipe.rb

```diff
@@ -69,37 +71,32 @@ def prepare_pipe
       end
 
       def run_pipe
-        @@pipeproc = Proc.new {
-          trap("INT") do
-            Process.exit
-          end
-          rundir = Dir.pwd
+        rundir = Dir.pwd
+        while @done==false do
           IO.popen("cd #{rundir} && #{SC.sclang_path.chomp} -d #{rundir.chomp} -i scvim", "w") do |sclang|
-            loop {
-              x = `cat #{@@pipe_loc}`
-              sclang.print x if x
-            }
```

Comment:
This while-loop does the same thing as the `cat` based code above, but without a seperate process that might not get cleaned up.